mocks/mock_pbft_failure_detector.hpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (C) 2018 Bluzelle
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License, version 3,
+// as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+#pragma once
+
+#include <pbft/pbft_failure_detector_base.hpp>
+#include <gmock/gmock.h>
+#include <functional>
+
+namespace bzn
+{
+    class Mockpbft_failure_detector_base : public pbft_failure_detector_base
```

Comment:
Was this handcrafted? If so then you may want to in the future use gmock_gen.y located:

./googletest/src/googletest/googlemock/scripts/generator/gmock_gen.py